File: aoj/volume100/java/A10032.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;

class A10032 {
    public static void main(String[] args) {
        Deque<String> stack = new ArrayDeque<String>();

        Scanner scan = new Scanner(System.in);
        while (true) {
            String instruction = scan.next();
            String block;

            switch (instruction) {
            case "quit":
                return;

            case "push":
                block = scan.next();
                stack.push(block);
                break;

            case "pop":
                block = stack.pop();
                System.out.println(block);
                break;
            }
        }
    }
}

File: aoj/volume100/php/A10032.php
<?php

$stack = [];

while (true) {
    fscanf(STDIN, "%s %s", $instruction, $block);
    if ($instruction === "quit")
        break;

    switch ($instruction) {
    case "push":
        array_push($stack, $block);
        break;
    case "pop":
        $block = array_pop($stack);
        echo "$block\n";
    }
}
